src/Joonasw.AspNetCore.SecurityHeaders/FeaturePolicy/FeaturePolicyOptions.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using Joonasw.AspNetCore.SecurityHeaders.FeaturePolicy.Options;
+
+namespace Joonasw.AspNetCore.SecurityHeaders.FeaturePolicy
+{
+    public class FeaturePolicyOptions
+    {
+        public FeaturePolicyOptions()
+        {
+            GeolocationOptions = new FeaturePolicyGeolocationOptions();
+            MidiOptions = new FeaturePolicyMidiOptions();
+            NotificationsOptions = new FeaturePolicyNotificationsOptions();
+            PushOptions = new FeaturePolicyPushOptions();
+            SyncXhrOptions = new FeaturePolicySyncXhrOptions();
+            MicrophoneOptions = new FeaturePolicyMicrophoneOptions();
+            CameraOptions = new FeaturePolicyCameraOptions();
+            MagnetometerOptions = new FeaturePolicyMagnetometerOptions();
+            GyroscopeOptions = new FeaturePolicyGyroscopeOptions();
+            SpeakerOptions = new FeaturePolicySpeakerOptions();
+            VibrateOptions = new FeaturePolicyVibrateOptions();
+            FullscreenOptions = new FeaturePolicyFullscreenOptions();
+            PaymentOptions = new FeaturePolicyPaymentOptions();
+        }
+
+        public FeaturePolicyGeolocationOptions GeolocationOptions { get; set; }
+        public FeaturePolicyMidiOptions MidiOptions { get; set; }
+        public FeaturePolicyNotificationsOptions NotificationsOptions { get; set; }
+        public FeaturePolicyPushOptions PushOptions { get; set; }
+        public FeaturePolicySyncXhrOptions SyncXhrOptions { get; set; }
+        public FeaturePolicyMicrophoneOptions MicrophoneOptions { get; set; }
+        public FeaturePolicyCameraOptions CameraOptions { get; set; }
+        public FeaturePolicyMagnetometerOptions MagnetometerOptions { get; set; }
+        public FeaturePolicyGyroscopeOptions GyroscopeOptions { get; set; }
+        public FeaturePolicySpeakerOptions SpeakerOptions { get; set; }
+        public FeaturePolicyVibrateOptions VibrateOptions { get; set; }
+        public FeaturePolicyFullscreenOptions FullscreenOptions { get; set; }
+        public FeaturePolicyPaymentOptions PaymentOptions { get; set; }
+
+        internal enum FeaturePolicyValues
```

Comment:
These perfectly match what Scott Helme wrote about at https://scotthelme.co.uk/a-new-security-header-feature-policy/ which is good :)

However, the "companion document" mentioned in the spec mentions other values: https://github.com/WICG/feature-policy/blob/master/features.md. So I am not sure which values we should even include. You can see it referred to from here: https://wicg.github.io/feature-policy/#features.

Reply:
Yeah good point.  I went ahead and added the rest.